src/main/java/com/exactpro/th2/codec/qfj/QFJCodec.java

```diff
@@ -0,0 +1,473 @@
+            listValue.addValues(ValueUtils.toValue(innerMessage));
+        }
+    }
+
+    private Message getMessage(Iterator<Field<?>> iterator, Message.Builder messageBuilder, DataDictionary dataDictionary, Group group, String msgType) {
+
+        while (iterator.hasNext()) {
+            Field<?> field = iterator.next();
+
+            if (dataDictionary.isGroup(msgType, field.getTag())) {
+                ListValue.@NotNull Builder listValue = ValueUtils.listValue();
+                List<Group> groups = group.getGroups(field.getTag());
+
+                DataDictionary.GroupInfo groupInfo = Objects.requireNonNull(dataDictionary.getGroup(msgType, field.getTag()),
+                        () -> "No GroupInfo for this combination of tag:{}" + field.getTag() + " and msgType:{}" + msgType);
+
+                DataDictionary innerDataDictionary = groupInfo.getDataDictionary();
+
+                fillListValue(listValue, innerDataDictionary, groups, msgType);
+                messageBuilder.putFields(localDataDictionary.getFieldName(field.getTag()), Value.newBuilder().setListValue(listValue).build());
+            } else {
+                if (!group.isSetField(field.getField())) {
+                    throw new IllegalArgumentException("Invalid tag \"" + dataDictionary.getFieldName(field.getField()) + "\" for message group " + group);
+                }
+                putMessageField(messageBuilder, localDataDictionary, field);
+            }
+        }
+        return messageBuilder.build();
+    }
+
+    //ADDING HEADER TO MESSAGE
+    private Message getMessage(Iterator<Field<?>> iterator, Message.Builder messageBuilder, quickfix.Message.Header header, boolean isFix50) {
+
+        DataDictionary transportDictionary = isFix50 ? transportDataDictionary : dataDictionary;
+
+        while (iterator.hasNext()) {
+            Field<?> field = iterator.next();
+
+            if (transportDictionary.isHeaderGroup(field.getTag())) {
+                ListValue.@NotNull Builder listValue = ValueUtils.listValue();
+                List<Group> groups = header.getGroups(field.getTag());
+
+                DataDictionary.GroupInfo groupInfo = Objects.requireNonNull(transportDictionary.getGroup(DataDictionary.HEADER_ID, field.getTag()),
+                        () -> "No GroupInfo for this combination of tag:{}" + field.getTag() + " and msgType:{}" + DataDictionary.HEADER_ID);
+                DataDictionary innerDataDictionary = groupInfo.getDataDictionary();
+
+                fillListValue(listValue, innerDataDictionary, groups, DataDictionary.HEADER_ID);
+                messageBuilder.putFields(transportDictionary.getFieldName(field.getTag()), Value.newBuilder().setListValue(listValue).build());
+            } else {
+                putMessageField(messageBuilder, transportDictionary, field);
+            }
+        }
+        return messageBuilder.build();
+    }
```

Comment:
This method looks almost exactly like a previous method. I think we can merge them into one